sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClientJavaDocCodeSnippets.java

```diff
@@ -264,115 +323,170 @@ public void beginRecognizeReceiptsFromUrl() {
-            .subscribe(recognizePollingOperation -> {
+        formRecognizerAsyncClient.beginRecognizeReceipts(buffer, receipt.length())
+            .flatMap(recognizePollingOperation ->
                 // if training polling operation completed, retrieve the final result.
-                recognizePollingOperation.getFinalResult().subscribe(recognizedReceipts -> {
-                    for (int i = 0; i < recognizedReceipts.size(); i++) {
-                        RecognizedReceipt recognizedReceipt = recognizedReceipts.get(i);
-                        Map<String, FormField> recognizedFields = recognizedReceipt.getRecognizedForm().getFields();
-                        System.out.printf("----------- Recognized Receipt page %s -----------%n", i);
-                        FormField merchantNameField = recognizedFields.get("MerchantName");
-                        if (merchantNameField.getFieldValue().getType() == FieldValueType.STRING) {
-                            System.out.printf("Merchant Name: %s, confidence: %.2f%n",
-                                merchantNameField.getFieldValue().asString(),
-                                merchantNameField.getConfidence());
-                        }
-                        FormField transactionDateField = recognizedFields.get("TransactionDate");
-                        if (transactionDateField.getFieldValue().getType() == FieldValueType.DATE) {
-                            System.out.printf("Transaction Date: %s, confidence: %.2f%n",
-                                transactionDateField.getFieldValue().asDate(),
-                                transactionDateField.getConfidence());
-                        }
-                        FormField receiptItemsField = recognizedFields.get("Items");
-                        System.out.printf("Receipt Items: %n");
-                        if (receiptItemsField.getFieldValue().getType() == FieldValueType.LIST) {
-                            List<FormField> receiptItems = receiptItemsField.getFieldValue().asList();
-                            receiptItems.forEach(receiptItem -> {
-                                if (receiptItem.getFieldValue().getType() == FieldValueType.MAP) {
-                                    receiptItem.getFieldValue().asMap().forEach((key, formField) -> {
-                                        if (key.equals("Quantity")) {
-                                            if (formField.getFieldValue().getType() == FieldValueType.INTEGER) {
-                                                System.out.printf("Quantity: %s, confidence: %.2f%n",
-                                                    formField.getFieldValue().asInteger(), formField.getConfidence());
-                                            }
+                recognizePollingOperation.getFinalResult())
+            .subscribe(recognizedReceipts -> {
+                for (int i = 0; i < recognizedReceipts.size(); i++) {
```

Comment:
Use `forEach()` instead or you can also show the reactor pattern here:
```java
formRecognizerAsyncClient.beginRecognizeReceipts(buffer, receipt.length())
     .flatMap(recognizePollingOperation -> recognizePollingOperation.getFinalResult())
     .flatMap(recognizedReceipts -> Flux.fromIterable(recognizedReceipts))
     .subscribe(recognizedReceipt -> {...});
```

Reply:
Yep, I am preparing a follow-up PR for sample and snippets update to follow reactor pattern more!